types.go

```diff
@@ -278,19 +282,43 @@ func (source *Source) Metadata() []MetadataField {
 
 func (source *Source) AuthenticateToECR() bool {
 	logrus.Warnln("ECR integration is experimental and untested")
-	mySession := session.Must(session.NewSession(&aws.Config{
-		Region:      aws.String(source.AwsRegion),
-		Credentials: credentials.NewStaticCredentials(source.AwsAccessKeyId, source.AwsSecretAccessKey, source.AwsSessionToken),
-	}))
 
-	var config aws.Config
+	var mySession *session.Session
 
-	// If a role arn has been supplied, then assume role and get a new session
-	if source.AwsRoleArn != "" {
-		config = aws.Config{Credentials: stscreds.NewCredentials(mySession, source.AwsRoleArn)}
+	if source.AwsEC2Credentials {
+		mySession = session.Must(session.NewSession(&aws.Config{
+			Region:      aws.String(source.AwsRegion),
+			Credentials: credentials.NewCredentials(&ec2rolecreds.EC2RoleProvider{Client: ec2metadata.New(session.New())}),
+		}))
+
+	} else {
+		mySession = session.Must(session.NewSession(&aws.Config{
+			Region:      aws.String(source.AwsRegion),
+			Credentials: credentials.NewStaticCredentials(source.AwsAccessKeyId, source.AwsSecretAccessKey, source.AwsSessionToken),
+		}))
+	}
+
+	// If aws role arns chain has been supplied, then assume roles in turn and get a new session
+	if source.AwsRoleArns != "" {
+		logrus.Warnln("Using `aws_role_arns` rather than `aws_role_arn`, see documentation")
+		awsRoleArnsList := strings.Split(source.AwsRoleArns, ",")
+
+		for _, roleArn := range awsRoleArnsList {
+			logrus.Debugf("assuming new role: %s", roleArn)
+			mySession = session.Must(session.NewSession(&aws.Config{
```

Comment:
Could we force these down the same codepath maybe? Something like:

```go
// Note: This implementation gives precedence to `aws_role_arn` since it
// assumes that we've errored if both are set (per my previous comment)
awsRoleArns := source.Aws.RoleArns
if source.AwsRoleArn {
	awsRoleArns = []string{source.AwsRoleArn}
}
for _, roleArn := range awsRoleArns {
	logrus.Debugf("assuming new role: %s", roleArn)
	mySession = session.Must(session.NewSession(&aws.Config{
		Region:      aws.String(source.AwsRegion),
		Credentials: stscreds.NewCredentials(mySession, roleArn),
	}
}
```

Also, I don't think we need the warning
> Using `aws_role_arns` rather than `aws_role_arn`, see documentation

if we disallow setting both

Reply:
cool, I can do that.